### PR TITLE
Enhance listVersionsFromCache behavior and documentation

### PR DESCRIPTION
**Pull request by Yantrio**
- Added detailed comments for the listVersionsFromCache function explaining its behavior.
- Improved handling for stale documents by triggering an asynchronous update and returning stale versions.
- Updated logging for clarity on both stale and non-stale cases.

This resolves an issue where timeouts were occuring fetching a large amount of versions when our cache was stale.

### CODE REVIEW

File: src/lambda/api/providerVersions.go [hunk 1]
```diff
@@ -92,10 +98,14 @@ func listVersionsFromCache(ctx context.Context, config config.Config, effectiveN
 	slog.Info("Found document in cache", "last_updated", document.LastUpdated, "versions", len(document.Versions))
 
 	if document.IsStale() {
-		return nil, nil
+		// if it's stale, trigger the lambda to update, and still return the stale document
+		slog.Info("Document is stale, returning cached versions and triggering lambda", "last_updated", document.LastUpdated)
+		if triggerErr := triggerPopulateProviderVersions(ctx, config, effectiveNamespace, providerType); triggerErr != nil {
```
Comment (RLRabinowitz) [inline src/lambda/api/providerVersions.go:103]: This works, but is a little confusing. `triggerPopulateProviderVersions` could happen here, if the cache is stale, or happen in the caller function `listProviderVersions` if there is not document (or fetching DDB cache errored)

**Review by RLRabinowitz — APPROVED** [review]
This should fix the issue, but is a bit confusing to me, regarding when we populate the DDB cache

File: src/lambda/api/providerVersions.go (hunk 1 above)
Comment (Yantrio) [reply src/lambda/api/providerVersions.go:103]: I fail to see how this is confusing. Yes there are a few paths to triggering the lambda. What do you propose as an alternative?

Comment (RLRabinowitz) [reply src/lambda/api/providerVersions.go:103]: It's just that the responsibility of triggering update for the cache is split up now. This method does more than just `listVersionsFromCache`

I thought it'd be clearer to have this function only return the document, and mention it's stale, and have the caller decide what to do with it. Would also make sure the responsibilities of each component here make more sense

But this is not a blocker at all IMO. Feel free to resolve this comment and merge

**Pull request merged**